nncf/common/pruning/mask_propagation.py

```diff
@@ -54,3 +61,74 @@ def mask_propagation(self):
         for node in self._graph.topological_sort():
             cls = self.get_meta_operation_by_type_name(node.node_type)
             cls.mask_propagation(node, self._graph)
+
+    def symbolic_mask_propagation(self, prunable_layers_types: List[str],
+                                  can_prune_after_analysis: Dict[int, PruningAnalysisDecision]) \
+            -> Dict[int, PruningAnalysisDecision]:
+        """
+        Check all nodes that were marked as prunable after the model analysis and compatibility check vs.
+        pruning algo have a correct correspondent closing node on each path form self to outputs;
+        the check entails verifying that every convolution prunable by the output channel dimension
+        has a corresponding convolution that is pruneable by its input channel dimension (output channel
+        dimension equal to closing convolution input channel dimension) in every path form self to outputs.
+        If the check fails, the entire groups containing such nodes will be marked as un-pruneable.
+        If convolution symbolic mask mixes with other symbolic masks (by elementwise operation, for example)
+        and mixing masks can't be mixed, all mask producers participated in this mixing will be marked as unprunable.
+        If convolution output channel dimension reducing directly affect an output of the model -
+        it will be marked as unprunable as well.
```

Comment:
```suggestion
        Check all nodes that were marked as prunable after the model analysis and compatibility check vs.
        pruning algo have a correct correspondent closing node on each path from self to outputs;
        the check entails verifying that every convolution prunable by the output channel dimension
        has a corresponding convolution that is prunable by its input channel dimension (output channel
        dimension equal to closing convolution input channel dimension) in every path form self to outputs.
        If the check fails, the entire groups containing such nodes will be marked as unprunable.
        If convolution symbolic mask mixes with other symbolic masks (by elementwise operation, for example)
        and mixing masks can't be mixed, all mask producers participated in this mixing will be marked as unprunable.
        If convolution output channel dimension reducing directly affect an output of the model -
        it will be marked as unprunable as well.
```